src/Serialization/SerializablePayload.php

```diff
@@ -8,5 +8,5 @@ interface SerializablePayload
 {
     public function toPayload(): array;
 
-    public static function fromPayload(array $payload): SerializablePayload;
+    public static function fromPayload(array $payload): self;
```

Comment:
Probably `static`?

Reply:
when using static on this interface you get into weird type shit where the internal constructor needs to use new static and a class name is no longer accepted. And because then you're open to late static binding, you have to make all your events final otherwise type errors. While all you want it for it to only return itself, so I think self articulates that intent the clearest.